src/rust/src/contract.rs

```diff
@@ -157,9 +157,9 @@ impl From<xdr::Error> for CoreHostError {
 impl std::error::Error for CoreHostError {}
 
 fn non_metered_xdr_from_cxx_buf<T: ReadXdr>(buf: &CxxBuf) -> Result<T, HostError> {
-    Ok(T::read_xdr(&mut xdr::DepthLimitedRead::new(
+    Ok(T::read_xdr(&mut xdr::Limited::new(
         Cursor::new(buf.data.as_slice()),
-        MARSHALLING_STACK_LIMIT,
+        Limits {depth: MARSHALLING_STACK_LIMIT, len: 0x1000000},
```

Comment:
Use the buffer size here for the length.

Reply:
fixed